hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/security/S3GetSecretRequest.java

```diff
@@ -167,7 +168,19 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
                 assignS3SecretValue = null;
               }
             } else {
-              // Found in S3SecretTable.
+              final OMMultiTenantManager multiTenantManager =
+                  ozoneManager.getMultiTenantManager();
+              if (multiTenantManager == null ||
+                  multiTenantManager.getTenantForAccessID(accessId) == null ||
+                  !multiTenantManager.getTenantForAccessID(accessId)
+                      .isPresent()) {
+                // Access Id is not assigned to any tenant and
+                // Secret is found in S3SecretTable. No secret is returned.
+                throw new OMException("Secret for '" + accessId +
```

Comment:
We should put this behind a config option. We should not break existing CLI behavior to avoid app breakage.

Reply:
Hi @kerneltime , we all know it's important to keep the backward compatibility for functions.  But this one, is a security flaw/issue in the secret management, which requires a fix/mitigation.  So, I don't think we should provide such an config option.

Reply:
It is not necessarily a flaw in the secret management. The retrieval of secrets is still protected by kerberos principal?